src/NextGen/CustomFields/ServiceProvider.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Give\NextGen\CustomFields;
+
+use Give\Framework\FieldsAPI\Field;
+use Give\Helpers\Hooks;
+use Give\NextGen\CustomFields\Controllers\DonationDetailsController;
+use Give\NextGen\CustomFields\Controllers\DonorDetailsController;
+use Give\ServiceProviders\ServiceProvider as ServiceProviderInterface;
+
+/**
+ * @unreleased
+ */
+class ServiceProvider implements ServiceProviderInterface
+{
+    /**
+     * @inheritDoc
+     */
+    public function register()
+    {
+        Field::macro('shouldDisplayInAdmin', function() {
```

Comment:
Woah, that's cool!

Reply:
Right? Macros are great here for extending the Fields API.

Reply:
So... I do love macros, but why not just bake this right into Fields? 😅

Reply:
That would require a change to Give core, which we can do, but I wanted to start here.